WordPress/Classes/ViewRelated/Blog/BlogDetailHeaderView.m

```diff
@@ -0,0 +1,116 @@
+
+@end
+
+@implementation BlogDetailHeaderView
+
+- (instancetype)initWithFrame:(CGRect)frame
+{
+    self = [super initWithFrame:frame];
+    if (self) {
+        _blavatarImageView = [self imageViewForBlavatar];
+        [self addSubview:_blavatarImageView];
+
+        _titleLabel = [self labelForTitle];
+        [self addSubview:_titleLabel];
+
+        _subtitleLabel = [self labelForSubtitle];
+        [self addSubview:_subtitleLabel];
+
+        [self configureConstraints];
+    }
+    return self;
+}
+
+#pragma mark - Public Methods
+
+- (void)setBlog:(Blog *)blog
+{
+    [self.blavatarImageView setImageWithBlavatarUrl:blog.blavatarUrl isWPcom:blog.isWPcom];
+
+    // if the blog name is missing, we want to show the blog displayURL instead
+    [self.titleLabel setText:(blog.blogName ? blog.blogName : blog.displayURL)];
+    [self.subtitleLabel setText:blog.displayURL];
+}
+
+#pragma mark - Private Methods
+
+- (void)configureConstraints
+{
+    NSDictionary *views = NSDictionaryOfVariableBindings(_blavatarImageView, _titleLabel, _subtitleLabel);
+    NSDictionary *metrics = @{@"blavatarSize": @(BlogDetailHeaderViewBlavatarSize),
+                              @"labelHeight":@(BlogDetailHeaderViewLabelHeight),
+                              @"labelHorizontalPadding": @(BlogDetailHeaderViewLabelHorizontalPadding)};
+    [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|[_blavatarImageView(blavatarSize)]|"
+                                                                 options:0
+                                                                 metrics:metrics
+                                                                   views:views]];
+    [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"|[_blavatarImageView(blavatarSize)]-labelHorizontalPadding-[_titleLabel]|"
+                                                                 options:0
+                                                                 metrics:metrics
+                                                                   views:views]];
+    [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"|[_blavatarImageView(blavatarSize)]-labelHorizontalPadding-[_subtitleLabel]|"
+                                                                 options:0
+                                                                 metrics:metrics
+                                                                   views:views]];
+    [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|[_titleLabel(labelHeight)][_subtitleLabel(labelHeight)]"
+                                                                 options:0
+                                                                 metrics:metrics
+                                                                   views:views]];
+    [super setNeedsUpdateConstraints];
+}
+
+#pragma mark - Subview factories
+
+- (UILabel *)labelForTitle
```

Comment:
No problem @oguzkocer!. 

Memory management aside (which is fine i think, ever since ARC took over), a good reason to use the prefix is: without looking at the implementation, you might think it's a simple getter (when it actually creates a new instance each time it's accessed).

Thanks!